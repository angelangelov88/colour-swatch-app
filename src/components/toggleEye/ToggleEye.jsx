import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { EyeIcon, EyeOffIcon } from '@heroicons/react/outline';

function ToggleEye(props) {
  const {
    className,
    disabled,
  } = props;

  const [active, setActive] = React.useState(true);
  const handleClick = () => {
    if (!disabled) {
      setActive(!active);
    }
  };

  const enterKeyPress = (e) => {
    if (e.key === 'Enter' || e.code === 'Space') {
      handleClick();
    }
  };

  const classes = classNames(
    'flex justify-center items-center group p-0.5 rounded-[10px] w-max text-gray-30 bg-gray-20',
    'hover:bg-primary-10',
    'focus:outline focus:outline-4 focus:outline-primary-30 focus:bg-primary-10',
    disabled && 'hover:cursor-default hover:bg-gray-20 focus:bg-gray-20',
    className,
  );

  const eyeClasses = classNames(
    'w-8 h-8 p-1.5',
    active && 'bg-white rounded-lg text-gray-90 shadow-sm',
    active && 'group-hover:shadow-sm',
    active && 'group-focus:shadow',
    (active && disabled) && 'text-gray-40 bg-gray-10 group-hover:shadow-none group-focus:shadow-none',
    (!active && disabled) && 'group-hover:text-gray-30 group-focus:text-gray-40',
    !active && 'group-hover:text-primary-light group-focus:text-primary-light',
  );

  const eyeOffClasses = classNames(
    'w-8 h-8  p-1.5',
    active && 'group-hover:text-primary-light',
    active && 'group-focus:text-primary-light',
    !active && 'bg-white rounded-lg text-gray-90 shadow-sm',
    !active && 'group-hover:shadow',
    !active && 'group-focus:shadow',
    (active && disabled) && 'group-hover:text-gray-30 group-focus:text-gray-40',
    (!active && disabled) && 'text-gray-40 bg-gray-10 group-hover:shadow-none group-focus:shadow-none',
  );

  return (
    <div className={classes} onClick={handleClick} role="button" tabIndex={0} onKeyDown={enterKeyPress}>
      <EyeIcon className={eyeClasses} />
      <EyeOffIcon className={eyeOffClasses} />
    </div>
  );
}

ToggleEye.defaultProps = {
  className: undefined,
  disabled: false,
};

ToggleEye.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
};

export default ToggleEye;
export { ToggleEye };
