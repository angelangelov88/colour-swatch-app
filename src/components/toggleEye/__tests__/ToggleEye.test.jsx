import React from 'react';
import renderer from 'react-test-renderer';
import { ToggleEye } from '../toggleEye';

it('should render with no props', () => {
  const component = renderer.create(
    <ToggleEye />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with disabled state', () => {
  const component = renderer.create(
    <ToggleEye disabled />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with red as bg colour', () => {
  const component = renderer.create(
    <ToggleEye className="bg-red-80" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
