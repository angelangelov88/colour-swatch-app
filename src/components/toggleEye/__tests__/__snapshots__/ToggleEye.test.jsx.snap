// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with disabled state 1`] = `
<div
  className="flex justify-center items-center group p-0.5 rounded-[10px] w-max text-gray-30 bg-gray-20 hover:bg-primary-10 focus:outline focus:outline-4 focus:outline-primary-30 focus:bg-primary-10 hover:cursor-default hover:bg-gray-20 focus:bg-gray-20"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    aria-hidden="true"
    className="w-8 h-8 p-1.5 bg-white rounded-lg text-gray-90 shadow-sm group-hover:shadow-sm group-focus:shadow text-gray-40 bg-gray-10 group-hover:shadow-none group-focus:shadow-none"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="w-8 h-8  p-1.5 group-hover:text-primary-light group-focus:text-primary-light group-hover:text-gray-30 group-focus:text-gray-40"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render with no props 1`] = `
<div
  className="flex justify-center items-center group p-0.5 rounded-[10px] w-max text-gray-30 bg-gray-20 hover:bg-primary-10 focus:outline focus:outline-4 focus:outline-primary-30 focus:bg-primary-10"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    aria-hidden="true"
    className="w-8 h-8 p-1.5 bg-white rounded-lg text-gray-90 shadow-sm group-hover:shadow-sm group-focus:shadow"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="w-8 h-8  p-1.5 group-hover:text-primary-light group-focus:text-primary-light"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;

exports[`should render with red as bg colour 1`] = `
<div
  className="flex justify-center items-center group p-0.5 rounded-[10px] w-max text-gray-30 bg-gray-20 hover:bg-primary-10 focus:outline focus:outline-4 focus:outline-primary-30 focus:bg-primary-10 bg-red-80"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    aria-hidden="true"
    className="w-8 h-8 p-1.5 bg-white rounded-lg text-gray-90 shadow-sm group-hover:shadow-sm group-focus:shadow"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M2.458 12C3.732 7.943 7.523 5 12 5c4.478 0 8.268 2.943 9.542 7-1.274 4.057-5.064 7-9.542 7-4.477 0-8.268-2.943-9.542-7z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="w-8 h-8  p-1.5 group-hover:text-primary-light group-focus:text-primary-light"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.88 9.88l-3.29-3.29m7.532 7.532l3.29 3.29M3 3l3.59 3.59m0 0A9.953 9.953 0 0112 5c4.478 0 8.268 2.943 9.543 7a10.025 10.025 0 01-4.132 5.411m0 0L21 21"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
</div>
`;
